packages/web-components/@sfx/products/test/unit/products.test.ts

```diff
@@ -69,21 +98,148 @@ describe('Products Component', () => {
   });
 
   describe('setProductsFromEvent', () => {
-    it('should set the event products payload into the component', () => {
-      const products = [1, 2, 3];
+    let products;
+    let group;
+
+    beforeEach(() => {
+      products = [1, 2, 3];
+      group = 'group';
+    });
+
+    it('should set products to an empty array if the event payload does not contain products', () => {
+      const event = { detail: {} };
+      component.setProductsFromEvent(event);
+
+      expect(component.products).to.deep.equal([]);
+    });
+
+    it('should set products when the event matches the group in the component', () => {
+      const event = { detail: { products, group } };
+      component.group = group;
+
+      component.setProductsFromEvent(event);
+
+      expect(component.products).to.equal(products);
+    });
+
+    it('should not set products when the group in the component and event do not match', () => {
+      const event = { detail: { products, group } };
+
+      component.setProductsFromEvent(event);
+
+      expect(component.products).to.deep.equal([]);
+    });
+
+    it('should default the group in the event to an empty string if it is falsey', () => {
+      const event = { detail: { products } };
+
+      component.setProductsFromEvent(event);
+
+      expect(component.products).to.equal(products);
+    });
+
+    it('should default the group in the component to an empty string if it is falsey', () => {
+      component.group = undefined;
       const event = { detail: { products } };
 
       component.setProductsFromEvent(event);
 
       expect(component.products).to.equal(products);
     });
+  });
+});
+
+describe('Products Search Component', () => {
+  let component;
+
+  beforeEach(() => {
+    component = new ProductsSearch();
+  });
+
+  describe('connectedCallback', () => {
+    it('should call super', () => {
+      const superConnected = stub(Object.getPrototypeOf(component), 'connectedCallback');
+
+      component.connectedCallback();
+
+      expect(superConnected).to.be.calledOnce;
+    });
+
+    it('should set up event listener for a provide-products event to set products', () => {
```

Comment:
Should the event be 'SEARCH_RESPONSE'? Or, where is the 'provide-products' wording coming from?